File: bay-leaf-example/src/main/webapp/bay-leaf-example/src/model/trade.js

class TradeRequest {

    constructor(id, symbol, quantity, price, side) {
        this.id=  id;
        this.symbol = symbol;
        this.quantity = quantity;
        this.price = price;
        this.side = side;
    }
}

class TradeResponse {
    constructor(id, isDone, rejectionMsg) {
        this.id = id;
        this.isDone = isDone;
        this.rejectionMsg = rejectionMsg;
    }
}

class TradeBlotterRequest {
    constructor() {

    }
}
export {TradeRequest, TradeResponse, TradeBlotterRequest };

File: bay-leaf-example/src/main/java/com/nikoskatsanos/bayleaf/netty/example/model/Trade.java
package com.nikoskatsanos.bayleaf.netty.example.model;

import com.nikoskatsanos.bayleaf.netty.example.model.TradeRequest.Side;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Data
public class Trade {

    private final long timestamp;
    private final String id;
    private final String symbol;
    private final double quantity;
    private final double price;
    private final Side side;
}
